presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkAdaptiveQueryExecution.java

```diff
@@ -254,11 +244,11 @@ public Void apply(Try<MapOutputStatistics> result)
                 else {
                     log.info("Fragment %s will not get executed now either because there was no exchange involved (a broadcast is present) or because of an unknown issue.",
                             fragment.getFragment().getId());
+                    publishFragmentCompletionEvent(new FragmentCompletionSuccessEvent(currentFragment.getFragment().getId(), Optional.empty()));
```

Comment:
why do we publish a fragmentcompletionEvent if no fragment was executed

Reply:
We do that to mark non-shuffle stages as ready for planning. Without this - event polling blocks for success/failure event of subtask which will not be executed if there was no shuffle involved